### DISCUSSION TITLE
Circular interpolation commands (G2 / G3) does not work synchronously with A axis

### DISCUSSION THREAD

**Discussion by takaozando** [grblHAL/core#411] [Q&A] [answered]
I have a 4 axis machine with the A axis as rotary. 
The linear commands move the axis synchronously when using G0 or G1 command passing all axis args. 
But when i try to send a circular interpolation command as G2 and G3, the machine do the circle then move the A axis after the circle move is done. There is any setting to make it synchronously?

**Comment by terjeio** [comment]
Gcode example please.

**Reply by takaozando** [reply]
G21 G54 G90 G40
G00 X-24.139 Y38.697 Z8
G02 X-24.139 Y38.697 A360 I25.798 J0 F500
M30

**Reply by terjeio** [reply]
Thanks. I do not have a 4-axis machine at hand but when running your code X, Y and A positions are updated in the DRO when the G2 block is executed and there are step pulses delivered to all three drivers.
Which driver/board combo are you using. And which grblHAL version? Use `$I` to get this information.

**Reply by takaozando** [reply]
We are using a version from 10/12/2021 with some modifications, this command may be returning some non-standard info.

[VER:1.1f.20210505:]
[OPT:NSL2,35,1024,5,0]
[NEWOPT:ENUMS,RT+,HOME,ES,TC,THC]
[FIRMWARE:grblHAL]
[NVS STORAGE:*EEPROM]
[DRIVER:STM32F401CC]
[DRIVER VERSION:210423]
[PLUGIN:THC v2.1.0 HASH: F646D48C]

**Reply by terjeio** [reply, marked as answer]
Ok, then either patch that version (at least [mc_arc()](https://github.com/grblHAL/core/blob/57243a0fc201fe4fdc97ad5880711a19eea49c0d/motion_control.c#L1C1-L395C1) must be updated, likely more) or update to the latest build and modify that.

**Comment by takaozando** [comment]
It worked just updating the mc_arc method, even removing the conditions below to avoid changes in more files. 

`if(!pl_data->condition.target_validated && grbl.check_arc_travel_limits) {`
`        pl_data->condition.target_validated = On;`
`        pl_data->condition.target_valid = grbl.check_arc_travel_limits((coord_data_t *)target, (coord_data_t *)position,`
`                                                                        (point_2d_t){ .x = (float)center.x, .y = (float)center.y },`
`                                                                         radius, plane, turns);`

Thank you so much!
